src/libhirte/common/time-util.c

```diff
@@ -20,3 +22,25 @@ uint64_t finalize_time_interval_micros(int64_t start_time_micros) {
 double micros_to_millis(uint64_t time_micros) {
         return (double) time_micros * microsec_to_millisec_multiplier;
 }
+
+int get_formatted_timestamp(char *timebuf_ret, const size_t buf_size) {
```

Comment:
This could be simplified (and renamed) to
```C
char* get_formatted_log_timestamp()
```
It would also slim down the code on the caller side. The `malloc`ed size is always the same and it should then be freed by the caller, of course. 
Its current (and probably only) purpose is to be used by the logging. Therefore, returning the error code is of little use - there won't be any logs if the logs have an error. Well, we could probably log without timestamp, but then failing to create info for the log causes one in addition to the actual log. Not sure if thats desired.

Reply:
Done